src/PanZoomWithCover.tsx

```diff
@@ -71,6 +71,15 @@ const PanZoomWithCover: React.FC<
       if (onCoverLoad) onCoverLoad();
     };
 
+    const intervalId = setInterval(() => {
+      if (image.naturalWidth > 0 && image.naturalHeight > 0) {
+        // naturalWidth and naturalHeight are all we need for scale calculations, so we can start
+        // progressively rendering the image
+        imageDimensionsKnownHandler();
+        clearInterval(intervalId);
+      }
+    }, 100);
+
     return () => {
       if (!panZoomRef.current) return;
```

Comment:
```suggestion
    const imageLoaded = false

    const imageLoadIntervalId = setInterval(() => {
      if (image.naturalWidth > 0 && image.naturalHeight > 0) {
        // naturalWidth and naturalHeight are all we need for scale calculations, so we can start
        // progressively rendering the image
        imageLoaded = true
        imageDimensionsKnownHandler();
        clearInterval(imageLoadIntervalId);
      }
    }, 100);
    
    // In case if image gets loaded sooner (for e.g. from cache)
    const onImageLoad = () => {
        if (imageLoaded) return
        
        imageDimensionsKnownHandler();
        clearInterval(imageLoadIntervalId);
    }
    
    image.addEventListener('load', onImageLoad)

    return () => {
      clearInterval(imageLoadIntervalId);
      image.removeEventListener('load', onImageLoad);

      if (!panZoomRef.current) return;
```

maybe we can leave onLoad but in a little bit different way. what do you think?

Reply:
Very good point! And thanks for getting back this quickly!

A few minor suggestions that do not alter the code's functionality but aim to enhance the readability and maintainability of the code. Maybe that's a question of personal preference - anyways, here's my perspective:

We can move both
        `imageLoaded = true`
and
        `if (imageLoaded) return`
inside `imageDimensionsKnownHandler`, which will make sure setting and checking is handled consistently, whoever code may change in the future.

Also, as the image might not have been fully loaded when `imageLoaded` is set to true, I would rename the flag to `imageAttached` (and `imageDimensionsKnownHandler` may be renamed to `attachImage` or `attachImageToDOM` or so).

Reply:
no problem:) you're right - `imageAttached` is a better name and renaming `imageDimensionsKnownHandler` to `attachImage` / `attachImageToDom` sounds good! and just like you said we can move `imageAttached` to function, should be cleaner